CRM/Core/Smarty.php

```diff
@@ -138,6 +138,12 @@ private function initialize() {
     }
 
     $this->register_function('crmURL', ['CRM_Utils_System', 'crmURL']);
+    if (CRM_Utils_Constant::value('CIVICRM_SMARTY_DEFAULT_ESCAPE')) {
+      if (!isset($this->_plugins['modifier']['escape'])) {
+        $this->register_modifier('escape', ['CRM_Core_Smarty', 'escape']);
+      }
+      $this->default_modifiers[] = 'escape:"htmlall"';
```

Comment:
Sorry what I meant was suppose `$this->default_modifiers` already contains something like `escape:"html"`. Then everything would end up with double-escaping.

Reply:
@demeritcowboy but then why would you add the define if you had some other method adding in a default modifier?

Reply:
Ahh.